File: 240308/원 모양으로 되어있는 방/a-room-in-a-circle.py
# 각 방부터 시작할 경우 
# 반시계 방향으로 돌면서 N-1번 이동할 때 각각 이동 합 구하기 

import sys
min_dist = sys.maxsize

n = int(input())
arr = [int(input()) for _ in range(n)]

for i in range(n):
    sum_dist = 0
    for j in range(n-1):
        nextIndex = (i+1+j) % n
        sum_dist += arr[nextIndex] * (j+1)

    min_dist = min(min_dist, sum_dist)

print(min_dist)

File: 240319/원 모양으로 되어있는 방/a-room-in-a-circle.js
const fs = require('fs')
const [n, ...arr] = fs.readFileSync(0).toString().trim().split('\n')

const N = +n
const rooms = arr.map(Number)
let minDist = Number.MAX_SAFE_INTEGER

for (let i=0;i<N;i++) {
    let sumDist = 0
    for (let j=0;j<N;j++) {
        if (i === j) continue

        const dist = (j - i + N) % N
        sumDist += dist * rooms[j]
    }

    minDist = Math.min(minDist, sumDist)
}

console.log(minDist)